### ISSUE TITLE
Binary support

### ISSUE THREAD

**Issue by Downchuck** [hmalphettes/redisdown#17]
To allow for binary support:
Simply need an argument to options, and/or  if(typeof(v) == "object") in the get command.
redis client takes: return_buffers: true.

**Comment by hmalphettes** [comment]
Thanks @Downchuck !
Would you be so kind to make a pull request please?

**Comment by tounano** [comment]
@hmalphettes I have same problem. I was able to solve it with 3 changes.

1. "return_buffers" should be default value...
2. On line 212 at iterator.js you should do: String(vkey), same as you're doing with value.

I can do pull request, do you want me to? Should I update package.json as well?

**Comment by hmalphettes** [comment]
Fantastic, yes please a PR is most welcome.
No need to worry about updating package.json, I will do that on the next `npm publish`

**Comment by tounano** [comment]
Hey,

Please check my pull request.

**Comment by tounano** [comment]
I found another bug:

Here's the thing:

When you store buffers in Redis, we don't know how they serialize them.

That's why when we're converting a buffer to string in NodeJS, and querying redis by that string, it doesn't work.

The solution is:

* If the key is buffer, we convert the prefixed strings to buffers.
* If the keys is a string, we proceed as usual.

That way, when we issue a request to Redis, even with a lua script, we let it convert the key by itself.

So in case you're prefixing strings to a key by a user in other places, please use `concatKey(string, string/buffer)`;

I looked for such places, but didn't find....

Cheers...

**Comment by tounano** [comment]
@hmalphettes it passes all tests now, when do you think you can publish it on NPM?

**Comment by hmalphettes** [comment]
@tounano thanks for all this work. Please give me a day to review this.

**Comment by tounano** [comment]
@hmalphettes 

I fixed more stuff...

Basically the problem with storing buffers as values happens because you perform json serialization on values.

I deleted all the instances of JSON serialization. Levelup is responsible for serializing keys/values into strings or buffers.

Redisdown should be responsible only for persisting it.

I can't think of a usecase where levelup instance would request persisting a JSON value from a leveldown instance.

Anyway, I deleted all the json serialization stuff.

It passes all tests.
